src/api/bulk2.ts

```diff
@@ -270,15 +270,14 @@ export class BulkV2<S extends Schema> {
     if (!options.pollTimeout) options.pollTimeout = this.pollTimeout;
     if (!options.pollInterval) options.pollInterval = this.pollInterval;
 
-    const job = this.createJob({
-      object: options.object,
-      operation: options.operation,
-    });
+    const { pollInterval, pollTimeout, input, ...createJobOpts } = options;
+
+    const job = this.createJob(createJobOpts);
```

Comment:
see the `NewIngestJobOptions` type above, it has some properties that can be set by the user but weren't passed down to `createJob` (`lineEnding` was the one I needed for the tests to pass, see my other comment: https://github.com/jsforce/jsforce/pull/1412/files#r1578340035).